plans: drop beakerlib from provision/virtual.py

The-Mule opened the pull request:
Installing beakerlib to the VM images is not really needed. It is actually breaking since if a test wants to use image that is not Fedora then this steps fails. For instance, to get beakerlib in RHEL or CentOS-Stream you need EPEL. IMHO if a virtual test requires beakerlib there is now much better way to get it - EPEL prepare feature in the actual test plan. 

Finally, notice that removing this part 

1. breaks nothing at the moment:

```
#  tmt -c how=provision run -vv -a plan -n /plans/provision/virtual/prepare
...
/var/tmp/tmt/run-002
Found 1 plan.

/plans/provision/virtual/prepare
summary: Virtual machine via testcloud
    discover
        how: fmf
        directory: /root/tmt
        hash: 9e261901
        filter: tag:provision-virtual
        tests: ^/tests/prepare
        summary: 4 tests selected
            /tests/prepare/ansible
            /tests/prepare/install
            /tests/prepare/multihost
            /tests/prepare/recommend
    ...
    execute
        queued execute task #1: default-0 on default-0

        execute task #1: default-0 on default-0
        how: tmt
        ignore-duration: false
        exit-first: false
            test: Check that ansible prepare works for basic images
                cmd: ./test.sh
                00:10:29 pass /tests/prepare/ansible (on default-0) [1/4]
            test: Exercise prepare/install plugin
                cmd: ./test.sh
                00:46:46 pass /tests/prepare/install (on default-0) [2/4]
            test: Check that requirement installation is guest-aware in multihost scenario
                cmd: ./test.sh
                00:02:05 pass /tests/prepare/multihost (on default-0) [3/4]
            test: Check that recommended packages work as expected
                cmd: ./test.sh
                00:05:54 pass /tests/prepare/recommend (on default-0) [4/4]

        summary: 4 tests executed

total: 4 tests passed

```

2. Unblocks testing of #3344.

Pull Request Checklist

* [ ] implement the feature
* [ ] write the documentation
* [ ] extend the test coverage
* [ ] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [ ] mention the version
* [ ] include a release note

lukaszachy commented:
> IMHO if a virtual test requires beakerlib there is now much better way to get it - EPEL prepare feature in the actual test plan

Purpose of that snippet is to avoid getting dnf metadata and installing beakerlib for each VM.

How about making sure this happens only for rhel/fedora/centos, excluding image-mode images?

The-Mule commented:
> How about making sure this happens only for rhel/fedora/centos, excluding image-mode images?

I guess we can do that. Let me try. We don't have RHEL images in the test suite right now IMO so it is essentially just about Fedora and CentOS Stream.

The-Mule commented:
Addressed @lukaszachy remark, update commit, PR summary and description.

See below (added echo $image for more verbose output, not included in this PR):

```
# tmt  --feeling-safe -c how=provision run -vvv -a plan -n /plans/provision/virtual/prepare
...
        prepare task #6: prepare-image on default-0
        how: shell
        summary: Fetch the image, refresh dnf cache, install beakerlib
        name: prepare-image
        order: 50
        overview: 1 script found
        script:
            tmt run --remove plan --default provision --how virtual finish
            for image in /var/tmp/tmt/testcloud/images/*qcow2; do
                echo $image
                if [[ $image =~ (Fedora|CentOS-Stream) ]]; then
                    if [[ $image =~ CentOS-Stream ]]; then
                      virt-customize --add $image --run-command 'dnf install -y epel-release --enablerepo="crb"'
                    fi
                    virt-customize --add $image --run-command 'dnf --refresh install -y beakerlib'
                fi
            done
            cmd: /var/tmp/tmt/run-034/plans/provision/virtual/prepare/tree/tmt-prepare-wrapper.sh-prepare-image-default-0
            err: /var/tmp/tmt/run-035
            err:
            err: /default/plan
            err:     provision
            err:         queued provision.provision task #1: default-0
            err:
            err:         provision.provision task #1: default-0
            err:         how: virtual
            err:         memory: 2048 MB
            err:         disk: 40 GB
            err:         progress: booting...
            err:         multihost name: default-0
            err:         arch: x86_64
            err:         distro: Fedora Linux 41 (Cloud Edition)
            err:
            err:         summary: 1 guest provisioned
            err:     finish
            err:
            err:         summary: 0 tasks completed
            err:         guest: stopped
            err:         guest: removed
            out: /var/tmp/tmt/testcloud/images/CentOS-Stream-GenericCloud-10-20250317.0.x86_64.qcow2
            out: [   0.0] Examining the guest ...
            out: [  12.6] Setting a random seed
            out: [  12.6] Running: dnf install -y epel-release --enablerepo="crb"
            out: [  14.8] SELinux relabelling
            out: [  33.9] Finishing off
            out: [   0.0] Examining the guest ...
            out: [  13.0] Setting a random seed
            out: [  13.0] Running: dnf --refresh install -y beakerlib
            out: [  15.7] SELinux relabelling
            out: [  34.4] Finishing off
            out: /var/tmp/tmt/testcloud/images/CentOS-Stream-GenericCloud-9-20250317.0.x86_64.qcow2
            out: [   0.0] Examining the guest ...
            out: [  12.3] Setting a random seed
            out: [  12.3] Running: dnf install -y epel-release --enablerepo="crb"
            out: [  15.5] SELinux relabelling
            out: [  34.6] Finishing off
            out: [   0.0] Examining the guest ...
            out: [  12.7] Setting a random seed
            out: [  12.7] Running: dnf --refresh install -y beakerlib
            out: [  15.8] SELinux relabelling
            out: [  33.5] Finishing off
            out: /var/tmp/tmt/testcloud/images/Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2
            out: [   0.0] Examining the guest ...
            out: [  13.5] Setting a random seed
            out: [  13.5] Running: dnf --refresh install -y beakerlib
            out: [  15.9] SELinux relabelling
            out: [  34.5] Finishing off
            out: /var/tmp/tmt/testcloud/images/Fedora-Cloud-Base-Generic-Rawhide-20250320.n.0.x86_64.qcow2
            out: [   0.0] Examining the guest ...
            out: [  11.9] Setting a random seed
            out: [  11.9] Running: dnf --refresh install -y beakerlib
            out: [  29.0] SELinux relabelling
            out: [  45.1] Finishing off
            out: /var/tmp/tmt/testcloud/images/Fedora-Cloud-Base-Generic.x86_64-40-1.14.qcow2
            out: [   0.0] Examining the guest ...
            out: [  12.4] Setting a random seed
            out: [  12.4] Running: dnf --refresh install -y beakerlib
            out: [  15.8] SELinux relabelling
            out: [  33.2] Finishing off
            out: /var/tmp/tmt/testcloud/images/fedora-coreos-41.20250302.3.2-qemu.x86_64.qcow2
```

psss commented on plans/provision/virtual.fmf:
```diff
@@ -29,7 +29,12 @@ prepare+:
         for image in /var/tmp/tmt/testcloud/images/*qcow2; do
-            virt-customize --add $image --run-command 'dnf --refresh install -y beakerlib'
+            if [[ $image =~ (Fedora|CentOS-Stream) ]]; then
+                if [[ $image =~ CentOS-Stream ]]; then
```
I'm afraid that this change has no effect, because on the line 30 we only download the default `virtual` provision plugin image which is Fedora. So there will be no `CentOS-Stream` under `/var/tmp/tmt/testcloud/images/`. Plus I'd say that plans which need `epel` to be enabled should explicitly state so.

The-Mule replied:
Is that so? Can you take a look at https://github.com/teemtee/tmt/pull/3601#issuecomment-2740522954 above? It runs the following code:

```
script: |
        tmt run --remove plan --default provision --how virtual finish
        for image in /var/tmp/tmt/testcloud/images/*qcow2; do
            echo $image
            virt-customize --add $image --run-command 'dnf --refresh install -y beakerlib'
            if [[ $image =~ (Fedora|CentOS-Stream) ]]; then
                if [[ $image =~ CentOS-Stream ]]; then
                  ...
```

And it seems to iterate over all virtual images from https://github.com/teemtee/tmt/blob/main/tests/images.sh#L39.

I'd definitely agree that epel should better be enabled on demand only but without it there is no beakerlib on CentOS-Stream images.

happz commented:
@The-Mule is this PR still relevant?
